File: VbNet/SqlDataServer/Dispose.cs
using System;
namespace SqlDataServer
{ 
public abstract class ReleaseObj : IDisposable
{
    public new void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }
    protected virtual new void Dispose(bool Disposing)
    {
        if (Disposing)
        {
        }
    }
    ~ReleaseObj()
    {
        Dispose(false);
    }
}
}

File: VbNet/SqlDataServer/Dispose.vb
Option Explicit On 
Option Strict On
Public MustInherit Class ReleaseObj
    Implements IDisposable
    Public Overloads Sub Dispose() Implements IDisposable.Dispose
        Dispose(True)
        GC.SuppressFinalize(Me)
    End Sub
    Protected Overridable Overloads Sub Dispose(ByVal Disposing As Boolean)
        If Disposing Then
        End If
    End Sub
    Protected Overrides Sub Finalize()
        Dispose(False)
    End Sub
End Class